### ISSUE TITLE
Don't count rate limit on a certain response

### ISSUE THREAD

**Issue by itschasa** [alisaifee/flask-limiter#334]
I was wondering if there was a function or method I could use to stop a request being counted on the rate limit of that route.

**Comment by alisaifee** [comment]
There is a global [exempt](https://flask-limiter.readthedocs.io/en/latest/index.html#registering-exemptions-from-rate-limits) decorator you could use to wrap a function that decides when to skip a rate limit or not.

If you want to make it very specific to a single rate limit that is marked on a specific route you can use the [`exempt_when`](https://flask-limiter.readthedocs.io/en/latest/api.html#flask_limiter.Limiter.limit.params.exempt_when) parameter which takes a callable that returns a boolean to decide if the rate limit should be skipped for the request. 

If you're using default limits on the extension there is a parallel `default_limits_exempt_when` argument that you can pass to the Limiter constructor ([reference](https://flask-limiter.readthedocs.io/en/latest/configuration.html#RATELIMIT_DEFAULTS_EXEMPT_WHEN))

**Comment by itschasa** [comment]
Could I set a condition, if response_code == 400, for the rate limit not to be counted?

**Comment by alisaifee** [comment]
Yes, you can use the `deduct_when` parameter: https://flask-limiter.readthedocs.io/en/stable/recipes.html#customizing-rate-limits-based-on-response

**Comment by itschasa** [comment]
exactly what i'm looking for, thank you <3
